### PR TITLE
refactor: Updating & cleaning

### PR DESCRIPTION
**Pull request by rschristian**
## Description

This PR updates the template and makes it usable again

## Reason for Change

Most of the dependencies were very out-dated, with some like `node-sass` causing errors when trying to use the template. It also fell behind the other templates and was not longer matching the ecosystem as a whole. 

Sorry, I do realize this is a big PR with 17 files changed. There was just a lot that needed touching up and as many changes built off of each other this was likely simpler than opening 4-5 smaller PRs

### Changelog
* [x]  Update all dependencies to newest(-ish)
* [x]  Refactor tests to use `enzyme` instead of `identity-obj-proxy`
* [x]  Drop Sass - Old version was causing some issues and not everyone wants to use it. Plain CSS is better in this scenario.
* [x]  Switch template from EJS to HTML. May as well now that the CLI provides an HTML template to use
* [x]  Update ESLint by adding the Preact Config
* [x]  Update documentation where appropriate

### CODE REVIEW

File: template/src/components/hello-world/index.js [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
+import "./style.css";
 
-export default class App extends Component {
-  render(props) {
+export default function App(props) {
     return (
-      <div>
-        <h1 style={{ color: props.color }}>Hello, World!</h1>
-      </div>
+        <div>
```
Comment (ForsakenHarmony) [inline template/src/components/hello-world/index.js]: This seems to be the only file which is indented by 4 spaces now

**Review by ForsakenHarmony — APPROVED** [review]

File: template/src/components/hello-world/index.js (hunk 1 above)
Comment (rschristian) [reply template/src/components/hello-world/index.js]: Very true. Sorry about that. Should be fixed now.

The only other file which may stand out is the `template.html` as it uses tabs for spacing. That was just copied straight from the CLI though, so I assume it's fine. Just as easy to fix too.

**Comment by ForsakenHarmony** [comment]
Hmm, not sure I understand why you put the widget specific code into `/demo` what was your thinking behind that?

**Comment by rschristian** [comment]
@ForsakenHarmony What in `/demo` is widget-specific? Should just be a method of consuming. 

The idea to split out was from the dozen or so conversations I've had on the Slack and the CLI repo in which so many users are completely lost as to what is what. There's a lot of missing clarity in what is just part of the demo and what is the actual source for the widget. Intermingling the two lead to a lot of confused users. 

Hopefully this should reduce if not alleviate that entirely. Seemed like it went over better to the users I talked to, but that's an anecdotal group of people/experience. Certainly open to suggestions.

**Comment by ForsakenHarmony** [comment]
Was thinking the only intended way to use a widget from this template would be embedding it with `preact-habitat`, not doing anything with the component directly

**Comment by rschristian** [comment]
And all those who want to do something with it directly? 

Not saying it's a use-case that has to be supported or anything but this way, IMO, both use cases are supported. Those who want to use habitat to get that sort of functionality have a working example and those that want to just make a component lib get something that makes sense. 

Just some thoughts; I'm certainly not married to anything.

**Comment by ForsakenHarmony** [comment]
Just not sure why you'd make a component lib with a widget template, then again I don't really think `preact-cli` is a good fit for either

**Comment by rschristian** [comment]
That might just depend on your definition of "widget". If we're just calling it an embeddable bit of JS that usually comes with a UI, I don't see much of a difference between the two. Sure, a component lib is a step further, requiring some way to render, but not far off itself. 

I do agree that `preact-cli` isn't a great fit, however, it allows for super easy scaffolding of templates which allows the Preact team to point at suggested ways of doing things. Until there's a better option for providing a similarly low-friction way of setting up repos I don't think it's the worst thing in the world? I think CLI is currently in a unique position to allow this sort of thing even if the template isn't directly for web applications. 

But again, just my opinions. Should I revert this + the TS to better resemble what was there? Then we at least can get something a bit more up-to-date in.

**Comment by ForsakenHarmony** [comment]
I don't have a strong opinion, can just leave as it is now

**Review by developit — APPROVED** [review]

**Comment by rschristian** [comment]
@ForsakenHarmony @developit 

I'm wondering if there's an argument to be made by having no external dependencies for the widget build? That might help to make the distinction between widget and component clearer. The ReadMe prior to this PR had "dist: Builds npm ready bundles in dist folder to be consumed by other preact web apps." so I figured I'd hold off on that for now as having `preact` be available in the consumer does seem to be the intention. I'd imagine a bundle with no external dependencies is far more desirable for a widget, but perhaps not? 

Regardless I changed this up a bit to better match what was talked about in the Slack, i.e., supporting both behaviors. I think it generally makes sense and should support everyone without going overboard.

File: template/package.json [hunk 2]
```diff
@@ -1,66 +1,48 @@
...
-    "preact-widget-scripts": "^0.6.0",
-    "sass-loader": "^6.0.6"
-  },
   "dependencies": {
-    "preact": "^8.5.2",
-    "preact-compat": "^3.17.0",
-    "preact-habitat": "^3.1.0",
-    "preact-render-to-string": "^4.1.0",
-    "preact-router": "^2.5.7"
+    "preact": "^10.5.7"
```
Comment (ForsakenHarmony) [inline template/package.json:28]: Should this be a peer dependency if used as a library?

File: template/tests/index.test.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { h } from "preact";
+import { shallow } from 'enzyme';
+
+import Hello from "../src/component";
+
+describe("Hello logic", () => {
+  test("should be able to run tests", () => {
+    expect(1 + 2).toEqual(3);
+  });
+});
```
Comment (ForsakenHarmony) [inline template/tests/index.test.js]: don't think we need this test :p

**Review by ForsakenHarmony — APPROVED** [review]
LGTM, feel free to merge

File: template/package.json (hunk 2 above)
Comment (rschristian) [reply template/package.json:28]: Ah, yeah, good point.

File: template/tests/index.test.js (hunk 3 above)
Comment (rschristian) [reply template/tests/index.test.js]: I'm hoping I copy/pasted that *wonderful and super useful* test from somewhere.... will remove it

**Comment by rschristian** [comment]
@ForsakenHarmony If that was aimed at me, I'm on a lower tier of permissions, no merge powers here 😅

**Pull request merged**